api-client/src/endpoints/createCategoryParentsEndpoint.js

```diff
@@ -0,0 +1,46 @@
+// @flow
+
+import CategoryModel from '../models/CategoryModel'
+import EndpointBuilder from '../EndpointBuilder'
+import type { JsonCategoryType } from '../types'
+import Endpoint from '../Endpoint'
+import mapCategoryJson from '../mapping/mapCategoryJson'
+import moment from 'moment-timezone'
+
+export const CATEGORY_PARENTS_ENDPOINT_NAME = 'categoryParents'
+
+type ParamsType = {| city: string, language: string, cityContentPath: string |}
+
+export default (baseUrl: string): Endpoint<ParamsType, Array<CategoryModel>> =>
+  new EndpointBuilder(CATEGORY_PARENTS_ENDPOINT_NAME)
+    .withParamsToUrlMapper((params: ParamsType): string => {
+      const { city, language, cityContentPath } = params
+      const basePath = `/${city}/${language}`
+      if (basePath === cityContentPath) {
+        throw new Error('This endpoint does not support the root category!')
+      }
+      return `${baseUrl}/${city}/${language}/wp-json/extensions/v3/parents?&url=${cityContentPath}`
+    })
+    .withMapper((json: Array<JsonCategoryType>, params: ParamsType): Array<CategoryModel> => {
+      const basePath = `/${params.city}/${params.language}`
+
+      const parents = json.map(category => mapCategoryJson(category, basePath))
+
+      parents.push(
+        new CategoryModel({
```

Comment:
I'm not sure whether it is wise to craft these weird CaregoryModels. We could also just make this endpoint return an array of paths. What do you think?

Reply:
We'd need at least the name and the path though to show breadcrumbs. We could do that though,yes. Or we just don't include the root at all. Not sure about that. On the other hand this just follows the behavior of the already existing categories endpoint, which is not really optimal though. So not sure.

Reply:
I think I'd leave this for now, we can refactor this at a later point of time together with the other endpoints using these weird models.